lib/resources/etc_hosts.rb

```diff
@@ -0,0 +1,82 @@
+# encoding: utf-8
+# author: Matthew Dromazos
+
+require 'utils/parser'
+require 'pry'
+class EtcHosts < Inspec.resource(1)
+  name 'etc_hosts'
+  desc 'Use the etc_hosts InSpec audit resource to find an
+    ip_address and its associated hosts'
+  example "
+    describe etc_hosts.where { ip_address == '127.0.0.1' } do
+      its ( 'ip_address' ) { should cmp '127.0.0.1' }
+      its ( 'primary_name' ) { should cmp 'localhost' }
+      its ( 'all_host_names' ) { should eq [['localhost', 'localhost.localdomain', 'localhost4', 'localhost4.localdomain4']] }
+    end
+  "
+
+  attr_reader :params
+
+  include CommentParser
+
+  def initialize(hosts_path = nil)
+    return skip_resource 'The `etc_hosts` resource is not supported on your OS.' unless inspec.os.linux? || inspec.os.windows?
+    @conf_path      = hosts_path || default_hosts_file_path(hosts_path)
+    @content        = nil
+    @params         = nil
+    read_content
+  end
+
+  filter = FilterTable.create
+  filter.add_accessor(:where)
+        .add_accessor(:entries)
+        .add(:ip_address,         field: 'ip_address')
+        .add(:primary_name, field: 'primary_name')
+        .add(:all_host_names,     field: 'all_host_names')
+
+  filter.connect(self, :params)
+
+  private
+
+  def default_hosts_file_path(hosts_path)
+    return hosts_path || '/etc/hosts' unless inspec.os.linux?
+    return hosts_path || 'C:\windows\system32\drivers\etc\hosts' unless inspec.os.windows?
+  end
+
+  def read_content
+    @content = ''
+    @params  = {}
+    @content = read_file(@conf_path)
+    @params  = parse_conf(@content)
+  end
+
+  def parse_conf(content)
+    content.map do |line|
+      data, = parse_comment_line(line, comment_char: '#', standalone_comments: false)
+      parse_line(data) unless data == ''
+    end.compact
+  end
+
+  def parse_line(line)
+    line_parts = line.split
+    return nil unless line_parts.length >= 2
+    {
+      'ip_address'         => line_parts[0],
```

Comment:
Can you please align these or remove the extraneous spaces? Lines 64-66. Either way is fine as long as it's consistent. Thanks!